color_color.py

```diff
@@ -0,0 +1,738 @@
+#!/usr/bin/env python
+
+#
+# Copyright (C) 2017, 2019 Smithsonian Astrophysical Observatory
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+
+"""Color Color Plot maker
+
+Creates a plot showing how model parameters affect hardness ratios
+given a user supplied ARF and RMF.
+
+"""
+
+import numpy as np
+import sherpa.astro.ui as ui
```

Comment:
This is a case where this is one of the things I've kept in my "back-pocket" for too long.    I have versions of this going back to 2017 and it looks like the sherpa read-the-docs stuff didn't come out until sherpa 4.10.1 in 2019.  Same w/ the diagonal rmf bit.

Reply:
My point about not using the ui layer is probably moot since I don't think we expect people to be importing this directly. But maybe it could be used in a srcflux plugin to do something .....?

Reply:
Oh, I see the ahelp file has an exlpicit examlpe of importing the module. That makes me think a bit more about this

Reply:
Yup, somebody once asked if this could be run in a notebook ... not that i'm naming names.

The import is only at the global level for convenience -- I could probably relocate it inside the ColorColor class.

From the user point-of-view, the "only" thing they need from the ui layer are the models.  I think that makes sense to keep this consistent with `modelflux` and `srcflux` and the other script I can't remember to make weighted instrument maps.

I'd like to think the only problem with using the `ui` interface is if a user also used `color_color` for their dataset id.